packages/office-ui-fabric-react/src/utilities/keytips/KeytipManager.ts

```diff
@@ -0,0 +1,430 @@
+    const uniqueKeytip: IUniqueKeytip = this._constructUniqueKeytip(keytipProps);
+
+    // Add keytip to the Manager, Tree, and Layer
+    // Check if trying to register a duplicate keytip, if so just update
+    const keytipIndex = this._findKeytipIndex(uniqueKeytip);
+    if (keytipIndex < 0) {
+      // Add
+      this.keytips.push(uniqueKeytip);
+      keytipTree.addNode(keytipProps, uniqueKeytip.uniqueID);
+      this._layer && this._layer.setKeytips(this.getKeytips());
+
+      // Add the keytip to the queue to show later
+      if (this.keytipTree.isCurrentKeytipParent(keytipProps)) {
+        this._addKeytipToQueue(convertSequencesToKeytipID(keytipProps.keySequences));
+      }
+    } else {
+      // Update
+      this._updateKeytip(uniqueKeytip, keytipIndex);
+    }
+
+    if (this._newCurrentKeytipSequences && fullKeySequencesAreEqual(keytipProps.keySequences, this._newCurrentKeytipSequences)) {
+      // This keytip should become the currentKeytip and should execute right away
+      let keytipSequence = [...keytipProps.keySequences];
+      if (keytipProps.overflowSetSequence) {
+        keytipSequence = mergeOverflowKeySequences(keytipSequence, keytipProps.overflowSetSequence);
+      }
+
+      // Set currentKeytip
+      keytipTree.currentKeytip = keytipTree.getNode(convertSequencesToKeytipID(keytipSequence));
+      if (keytipTree.currentKeytip) {
+        // Show all children keytips if there
+        const children = keytipTree.getChildren();
+        children.length && this.showKeytips(children);
+
+        if (keytipTree.currentKeytip.onExecute) {
+          keytipTree.currentKeytip.onExecute(this._getKeytipDOMElement(keytipTree.currentKeytip.id));
+        }
+      }
+
+      // Unset _newCurrentKeytipSequences
+      this._newCurrentKeytipSequences = undefined;
+    }
+
+    return uniqueKeytip.uniqueID;
+  }
+
+  /**
+   * Register a persisted keytip
+   * This means just adding a KeytipTreeNode
+   *
+   * @param keytipProps - Persisted Keytip to register
+   * @returns {string} Unique ID for this persisted keytip
+   */
+  public registerPersistedKeytip(keytipProps: IKeytipProps): string {
+    const uniqueID = this._getNextUniqueID();
+    this.keytipTree.addNode(keytipProps, uniqueID, true);
+    return uniqueID;
+  }
+
+  /**
+   * Update a keytip's props
+   *
+   * @param keytipProps - Keytip to update
+   * @param uniqueID - Unique ID of this keytip
+   */
+  public updateKeytip(keytipProps: IKeytipProps, uniqueID: string): void {
+    // Update keytip in this.keytips
+    const uniqueKeytip = this._constructUniqueKeytip(keytipProps, uniqueID);
+    const keytipIndex = this._findKeytipIndex(uniqueKeytip);
+    if (keytipIndex >= 0) {
+      this._updateKeytip(uniqueKeytip, keytipIndex);
+    }
+  }
+
+  /**
+   * Unregisters a keytip
+   *
+   * @param keytipToRemove - IKeytipProps of the keytip to remove
+   */
+  public unregisterKeytip(keytipToRemove: IKeytipProps, uniqueID: string): void {
```

Comment:
>unregisterKeytip [](start = 9, length = 16)

could you please expand on what the difference between this and a persistedKeytip?

Reply:
A persisted keytip is only used when an item with a keytip goes into an OverflowSet. The keytip path to that item will now include the overflow button sequence, so we add a persisted keytip to make a "shortcut" to that item without the overflow button sequence. This allows a user who is used to one sequence still be able to execute it (for example in the SLR, if they shrink their window the items go in the overflow, but the persisted keytip allows them to still trigger the original sequence as if it was still visible in the bar)